flight/tests/coordinate_conversions/unittest.cpp

```diff
@@ -85,3 +85,306 @@ TEST_F(RneFromLLATest, Equator) {
+  double dN = 0.0111319665326199;
+  double dE = 0.00822438930622829;
+  double dD = -1;
+
+  double T[3];
+  LLA_linearization_double(homeLLA_D[0], homeLLA_D[2], T);
+
+  float eps = 0.0000001f;
+  ASSERT_NEAR(dN/T[0], 1, eps);
+  ASSERT_NEAR(dE/T[1], 1, eps);
+  ASSERT_NEAR(dD/T[2], 1, eps);
+
+};
+
+TEST_F(LLALineraization, Sydney_double) {
+  // Test location is Sydney, AUS (According to wikipedia)
+  float homeLLA_D[] = {-33.86*1e7, 151.2*1e7, 10};
+  float dN = 0.0111319665326199;
+  float dE = 0.00924400128736662;
+  float dD = -1;
+
+  float T[3];
+  LLA_linearization_float(homeLLA_D[0], homeLLA_D[2], T);
+
+  float eps = 0.000001f;
+  ASSERT_NEAR(dN/T[0], 1, eps);
+  ASSERT_NEAR(dE/T[1], 1, eps);
+  ASSERT_NEAR(dD/T[2], 1, eps);
+
+};
+
+TEST_F(LLALineraization, Equator_on_ground) {
+  // Test location is Equator and Prime Meridian (According to wikipedia). This is just checking that everything works if
+  // there is no position change
+  float homeLLA_D[] = {0, 0, 0};
+  float currentLLA_D[] = {0, 0, 0};
+
+  float T[3];
+  LLA_linearization_float(homeLLA_D[0], homeLLA_D[2], T);
+
+  float NED[3];
+  get_linearized_3D_transformation(currentLLA_D[0], currentLLA_D[1], currentLLA_D[2],
+                                   homeLLA_D[0], homeLLA_D[1], homeLLA_D[2],
+                                   T, NED);
+  float eps = 0.000001f;
+  ASSERT_NEAR(0, NED[0], eps);
+  ASSERT_NEAR(0, NED[1], eps);
+  ASSERT_NEAR(0, NED[2], eps);
+};
+
+
+TEST_F(LLALineraization, Equator_to_Takeoff) {
+  // Test location is Equator and Prime Meridian (According to wikipedia)
+  double homeLLA_D[] = {0, 0, 0};
+  float currentLLA_D[] = {0, 0, 20};
+
+
+  float T[3];
+  LLA_linearization_float(homeLLA_D[0], homeLLA_D[2], T);
+
+  float NED[3];
+  get_linearized_3D_transformation(currentLLA_D[0], currentLLA_D[1], currentLLA_D[2],
+                                   homeLLA_D[0], homeLLA_D[1], homeLLA_D[2],
+                                   T, NED);
+
+  // 0.0110574275839048
+  // 0.0111319490810727
+  // -0.00100000016391277
+
+  float eps = 0.000001f;
+  ASSERT_NEAR(0, NED[0], eps);
+  ASSERT_NEAR(0, NED[1], eps);
+  ASSERT_NEAR(-20, NED[2], eps);
+
+};
+
+TEST_F(LLALineraization, Boston) {
+  // Test location is Boston, position change is smallest distance system can calculate. (true for following Aconagua
+  // and Challenger Deep tests
+  float homeLLA_D[] = {42.37*1e7, -71.03*1e7, 0};
+
+  int32_t dLat = 1;
+  int32_t dLon = 1;
+  float dAlt = 1;
+
+  int32_t currentLat = (int32_t)homeLLA_D[0] + dLat;
+  int32_t currentLon = (int32_t)homeLLA_D[1] + dLon;
+  float currentAlt   = homeLLA_D[2] + dAlt;
+
+  float T[3];
+  LLA_linearization_float(homeLLA_D[0], homeLLA_D[2], T);
+
+  float NED[3];
+  get_linearized_3D_transformation(currentLat, currentLon, currentAlt,
+                                   homeLLA_D[0], homeLLA_D[1], homeLLA_D[2],
+                                   T, NED);
+
+  float eps = 0.003f;
```

Comment:
So I guess this means that the linearization results in values that are only correct to 99.7%?